File: datasets/jl-keep-code_davinci_001_temp_0.2-0/HumanEval_25_factorize.jl
""" Return list of prime factors of given integer in the order from smallest to largest.
    Each of the factors should be listed number of times corresponding to how many times it appeares in factorization.
    Input number should be equal to the product of all factors
    >>> factorize(8)
    [2, 2, 2]
    >>> factorize(25)
    [5, 5]
    >>> factorize(70)
    [2, 5, 7]
    """
function factorize(n::Int64)::Vector{Int64} 
    factors = Int64[]
    for i in 2:n
        while n % i == 0
            push!(factors, i)
            n = div(n, i)
        end
    end
    return factors
end
factorize(8)
factorize(25)
factorize(70)

using Test

@testset begin

candidate = factorize;
	@test(candidate(2) == [2])
	@test(candidate(4) == [2, 2])
	@test(candidate(8) == [2, 2, 2])
	@test(candidate(57) == [3, 19])
	@test(candidate(3249) == [3, 3, 19, 19])
	@test(candidate(185193) == [3, 3, 3, 19, 19, 19])
	@test(candidate(20577) == [3, 19, 19, 19])
	@test(candidate(18) == [2, 3, 3])
end

File: datasets/pl-keep-code_davinci_001_temp_0.2/HumanEval_25_factorize.pl
# Return list of prime factors of given integer in the order from smallest to largest.
# Each of the factors should be listed number of times corresponding to how many times it appeares in factorization.
# Input number should be equal to the product of all factors
# >>> factorize(8)
# [2, 2, 2]
# >>> factorize(25)
# [5, 5]
# >>> factorize(70)
# [2, 5, 7]
sub factorize {
    my($n) = @_;
    my @factors = ();
    my $i = 2;
    while ($n > 1) {
        while ($n % $i == 0) {
            push @factors, $i;
            $n /= $i;
        }
        $i++;
    }
    return @factors;
}

use Test::Deep;


sub testhumaneval {
    my $candidate = \&factorize;
        if(eq_deeply($candidate->(2),[2])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(4),[2, 2])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(8),[2, 2, 2])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(57),[3, 19])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(3249),[3, 3, 19, 19])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(185193),[3, 3, 3, 19, 19, 19])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(20577),[3, 19, 19, 19])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->(18),[2, 3, 3])) {
        print "ok!" }else{
        exit 1;
        }
}

testhumaneval();